mixing/mixclient/client.go

```diff
@@ -431,6 +432,60 @@ func (c *Client) forLocalPeers(ctx context.Context, s *sessionRun, f func(p *pee
 	return errors.Join(errs...)
 }
 
+type delayedMsg struct {
+	t time.Time
+	m mixing.Message
+	p *peer
+}
+
+func (c *Client) sendLocalPeerMsgs(ctx context.Context, s *sessionRun, m func(p *peer) mixing.Message) error {
+	msgs := make([]delayedMsg, 0, len(s.peers))
+
+	now := time.Now()
+	for _, p := range s.peers {
+		if p.remote || p.ctx.Err() != nil {
+			continue
+		}
+		msg := m(p)
+		if p.triggeredBlame {
```

Comment:
it's more insurance than anything else, if we have marked our peer as triggering blame, we shouldn't be sending anything except an RS.

forLocalPeers runs the closure on separate worker goroutines but the workers have all completed before forLocalPeers returns.  There's no race (if there were, we would be racing on not just p.triggeredBlame here, but also each call to m to obtain the particular message for this stage).

Reply:
Some of those early returns after forLocalPeers returns errTriggeredBlame don't look 100% correct to me after taking a closer glance now. That prevents peers who did not observe the blaming situation to not send their non-RS messages at all.  These messages aren't needed for the actual blame function, but it could be used to correlate which peers belong to the same wallet (which is really what we are trying to avoid in this PR).

Reply:
should be fixed now, but will leave you to take another look

Reply:
Updates look good.